spec.html

```diff
@@ -18,6 +18,102 @@ <h1>Array Objects</h1>
+                    1. <del>If _y_ is *undefined*, return *-1*<sub>𝔽</sub>.</del>
+                    1. <del>If _comparefn_ is not *undefined*, then</del>
+                        1. <del>Let _v_ be ? ToNumber(? Call(_comparefn_, *undefined*, &laquo; _x_, _y_ &raquo;)).</del>
+                        1. <del>If _v_ is *NaN*, return *+0*<sub>𝔽</sub>.</del>
+                        1. <del>Return _v_.</del>
+                    1. <del>[id="step-sortcompare-tostring-x"] Let _xString_ be ? ToString(_x_).</del>
+                    1. <del>[id="step-sortcompare-tostring-y"] Let _yString_ be ? ToString(_y_).</del>
+                    1. <del>Let _xSmaller_ be ! IsLessThan(_xString_, _yString_, *true*).</del>
+                    1. <del>If _xSmaller_ is *true*, return *-1*<sub>𝔽</sub>.</del>
+                    1. <del>Let _ySmaller_ be ! IsLessThan(_yString_, _xString_, *true*).</del>
+                    1. <del>If _ySmaller_ is *true*, return *1*<sub>𝔽</sub>.</del>
+                    1. <del>Return *+0*<sub>𝔽</sub>.</del>
+                    1. <ins>Return ? ArraySortComparison(_x_, _y_, _comparefn_).</ins>
+                1. Return ? SortIndexedProperties(_obj_, <ins>_obj_</ins>, _len_, _SortCompare_, <ins>*true*</ins>).
+                </emu-alg>
+            </emu-clause>
+
+            <emu-clause id="sec-arraysortcomparison" type="abstract operation">
+                <h1>
+                  ArraySortComparison(
+                    _x_: unknown,
+                    _y_: unknown,
+                    _comparefn_: unknown
+                  ): either a normal completion containing a Number or an abrupt completion
+                </h1>
+                <dl class="header">
+                </dl>
+                <emu-alg>
+                    1. If _x_ and _y_ are both *undefined*, return *+0*<sub>𝔽</sub>.
+                    1. If _x_ is *undefined*, return *1*<sub>𝔽</sub>.
+                    1. If _y_ is *undefined*, return *-1*<sub>𝔽</sub>.
+                    1. If _comparefn_ is not *undefined*, then
+                        1. Let _v_ be ? ToNumber(? Call(_comparefn_, *undefined*, &laquo; _x_, _y_ &raquo;)).
+                        1. If _v_ is *NaN*, return *+0*<sub>𝔽</sub>.
+                        1. Return _v_.
+                    1. [id="step-sortcompare-tostring-x"] Let _xString_ be ? ToString(_x_).
+                    1. [id="step-sortcompare-tostring-y"] Let _yString_ be ? ToString(_y_).
+                    1. Let _xSmaller_ be ! IsLessThan(_xString_, _yString_, *true*).
+                    1. If _xSmaller_ is *true*, return *-1*<sub>𝔽</sub>.
+                    1. Let _ySmaller_ be ! IsLessThan(_yString_, _xString_, *true*).
+                    1. If _ySmaller_ is *true*, return *1*<sub>𝔽</sub>.
+                    1. Return *+0*<sub>𝔽</sub>.
+                </emu-alg>
+            </emu-clause>
+
+            <emu-clause id="sec-sortindexedproperties" type="abstract operation">
```

Comment:
`sortIndexedProperties` is updated to make it usable by `toSorted`.

a) toSorted needs to write out to a new array, rather than mutate the original instance
b) toSorted ignores 'holes' in arrays, and will read them regardless.